Development

Jamiec3466 opened the pull request.

Jamiec3466 committed:
Updated Code

Jamiec3466 committed:
Testing Update 2

Most of it seems to work except a "get / set" is required in the testing class but I have no clue why that is

finob2805 commented on CMP1903_A1_2324/Die.cs:
```diff
@@ -14,10 +14,19 @@ internal class Die
+        static Random rnd = new Random(); //RNG
+        public int RollTheDice() //Empty Integer value which will later be used to store the random value of the dice roll
+        {
+            NumRolled = rnd.Next(1, 7); //Uses the empty placeholder "NumRolles" to store the random value created in line 24 using 1 and 7 as the parameters
```
NumRolles typo

sebzera commented on CMP1903_A1_2324/Die.cs:
Code runs, does what is expected. Formatted well, concise and uses the right naming style. Comments are good

sebzera commented on CMP1903_A1_2324/Game.cs:
Code is concise and well commented, runs as it should, PascalCase is used and styling is good.

sebzera commented on CMP1903_A1_2324/Testing.cs:
Testing class does what it should do, and every other criteria is met

sebzera commented on CMP1903_A1_2324/Program.cs:
Game object creation is good, testing object creation is good. Concise and written well.

finob2805 commented on CMP1903_A1_2324/Die.cs:
Requirements have been met for the die class.
Code is easy to read and understand as comments have been used and variable names are informative
Code style matches guidelines
Typo on line 27 (NumRolles)

sebzera reviewed (commented):
Overall the code is written well, easy to read and is styled correctly. Functions as expected. Good job!

finob2805 commented on CMP1903_A1_2324/Game.cs:
Functionality is implemented - Three dice objects created, all 3 are rolled and sums are added
Code is easy to understand - Comments utilised throughout
Variable names are suitable

finob2805 commented on CMP1903_A1_2324/Program.cs:
Objects have been created and comments are used to show this
One syntax error at the end

finob2805 commented on CMP1903_A1_2324/Testing.cs:
Syntax errors are visible in this class
AllDieRolls() is called from the wrong class (line 28-29)
Debug.Assert() is not utilised for the testing
Comments make it easier to read

finob2805 reviewed (commented):
Overall most of the code is easily understandable and would run well. The only problems I saw were in the testing class where syntax errors prevented it from running, and that the Debug.Assert() function was not utilised so errors have not been handled correctly. Follows the coding guidelines.

Jamiec3466 committed:
Response to feedback

Syntax errors have been removed and Debug.Asset has been re-added; code appears to work now
